kubeadm: HA documentation (#12923)
kubernetes/website: content/en/docs/setup/independent/high-availability.md

    - `controlPlaneEndpoint` should match the address or DNS and port of the load balancer.
    - It's recommended that the versions of kubeadm, kubelet, kubectl and Kubernetes match.

1.  Initialize the control plane:

    ```sh
    sudo kubeadm init --config=kubeadm-config.yaml --experimental-upload-certs
    ```

>     - The `--experimental-upload-certs` flags is used to upload the control plane

Reviewer: What about:
"The `--experimental-upload-certs` flags is used to upload the certificates that should be shared across all the control-plane instances to the cluster; If instead, you prefer to copy certs across control-plane nodes manually or using automation tools, please remove this flag and refer to "manual certs distribution" paragraph"

  certificates to the cluster. 

    You should see something like:
    
    ```sh
    ...

>     You can now join any number of control-plane node running the following command on each as a root:

Reviewer 2: ```suggestion
    You can now join any number of control-plane nodes by running the following command on each as a root:
```

Reviewer 3: this is CLI typo, can't fix here.

      kubeadm join 192.168.0.200:6443 --token 9vr73a.a8uxyaju799qwdjv --discovery-token-ca-cert-hash sha256:7c2e69131a36ae2a042a339b33381c6d0d43887e2de83720eff5359e26aec866 --experimental-control-plane --certificate-key f8902e114ef118304e561c3ecd4d0b543adc226b7a07f675f56564185ffe0c07
    

>     Please note that the certificate-key gives access to cluster sensitive data, keep it secret!

Reviewer 2: Maybe highlight this a bit more by wrapping it in note tags

    As a safeguard, uploaded-certs will be deleted in two hours; If necessary, you can use kubeadm init phase upload-certs to reload certs afterward.
    
    Then you can join any number of worker nodes by running the following on each as root:
      kubeadm join 192.168.0.200:6443 --token 9vr73a.a8uxyaju799qwdjv --discovery-token-ca-cert-hash sha256:7c2e69131a36ae2a042a339b33381c6d0d43887e2de83720eff5359e26aec866
    ```

    - Copy this output to a text file. You will need it later to join control plane and worker nodes to the cluster.
    - When `--experimental-upload-certs` is used with `kubeadm init`, the certificates of the primary control plane

>     are uploaded to the cluster in a Secret and encrypted.

Reviewer 4: Maybe `are encrypted and uploaded to the kubeadm-certs secret`

>     -  When joining new control plane nodes using `kubeadm join ... --experimental-control-plane`,

Reviewer: Please move this point after the `kubeadm join --experimental-control-plane` command (L256 before your changes)

    kubectl get pod -n kube-system -w
    ```

>     - It's recommended that you join new control plane nodes in a sequence, only after the first node has finished initializing.

Reviewer: Please move this point before/after the kubeadm join --experimental-control-plane command (L256 before your changes)

>     - It's recommended that you join new control plane nodes in a sequence, only after the first node has finished initializing.

Reviewer 2: ```suggestion
    - It's recommended that you join new control plane nodes sequentially, only after the first node has finished initializing.
```

> 1.  Optionally copy the `admin.conf` file from the first control plane node to the rest:

Reviewer 4: Why we should copy this config? [1]

Reviewer 3: to use kubectl on the other CP nodes.
i'm OK with removing this but waiting on review by @fabriziopandini

Reviewer 4: The join don't generate a config?

Reviewer 3: that's actually a good point...
possibly best to remove this anyway.

@Klaven please remove the lines `1.  Optionally copy the `admin......` and `Optionally move the `admin.conf` file created....`

Reviewer: It is not necessary to copy the `admin.conf` file
However, I think we should still provide user instruction for manually copy certs if they want to opt out from the `--experimental-upload-certs` feature, so my suggestion is to move everyting related to "configure SSH" and scp certs (without the `admin.conf` file) in a dedicated paragraph "manual certs distribution"

    In the following example, replace `CONTROL_PLANE_IPS` with the IP addresses of the
    other control plane nodes.
    ```sh
    USER=ubuntu # customizable
    CONTROL_PLANE_IPS="10.0.0.7 10.0.0.8"
    for host in ${CONTROL_PLANE_IPS}; do
        scp /etc/kubernetes/admin.conf "${USER}"@$host:
    done
    ```

### Steps for the rest of the control plane nodes

> 1.  Optionally move the `admin.conf` file created by the previous step where `scp` was used:

Reviewer 4: [1]

        - `ETCD_1_IP`
        - `ETCD_2_IP`

> 1.  Run `kubeadm init --config kubeadm-config.yaml --experimental-upload-certs` on this node.

Reviewer: @yagonobre I don't remember if certificates for the external etcd are copied automatically or not...
If not, we should add a note that the user must take care of distributing those certs on all control-plane nodes

Reviewer 4: it's upload, we only skip certs upload in case of external ca.
